src/smhi/smhi.py

```diff
@@ -76,116 +66,145 @@ def find_stations_from_gps(
 
         Args:
             parameter: station parameter
             dist: distance from city
             city: name of city
+
+        Returns:
+            nearby stations
         """
         geolocator = Nominatim(user_agent="ifk-smhi")
         loc = geolocator.geocode(city)
-        self.find_stations_from_gps(
-            parameter=parameter,
-            dist=dist,
+        return self._find_stations_from_gps(
+            station_response,
             latitude=loc.latitude,
             longitude=loc.longitude,
+            dist=dist,
         )
 
     def get_data(
         self,
         parameter: int,
         station: int,
-        period: str = "corrected-archive",
-        interpolate: int = 0,
+        distance: int = 0,
     ) -> Tuple[Any, Any]:
         """Get data from station.
 
         Args:
             parameter: data parameter
             station: station id
-            period: period to get
+            distance: station distance
+
+        Returns:
+            data
         """
-        data, header = self.client.get_data_from_selection(
-            parameter=parameter, station=station, period=period
-        )
-        if interpolate > 0:
+        stations = Stations(Parameters(), parameter)
+        periods = Periods(stations, station)
+        data = Data(periods)
+
+        return self._interpolate(self, distance, stations, periods, data)
+
+    def get_data_by_city(
+        self,
+        parameter: int,
+        city: str,
+        distance: int = 0,
+    ) -> Tuple[Any, Any]:
+        """Get data from station.
+
+        Args:
+            parameter: data parameter
+            city: user provided city
+            distance: station distance
+
+        Returns:
+            data
+        """
+        stations = Stations(Parameters(), parameter)
+        station = self._find_stations_by_city(
+            Stations(Parameters(), parameter), city, distance
+        )[0]
+        periods = Periods(stations, station[0])
+        data = Data(periods)
+
+        return self._interpolate(self, distance, stations, periods, data)
+
+    def _interpolate(self, distance, stations, periods, data):
+        if distance > 0:
             # Find the station latitude and longitude information from Metobs
-            stat = next(
-                item for item in self.client.stations.stations if item["id"] == station
+            # should be replaced by a self.periods.position[0].latitude
+            latitude, longitude = (
+                periods.position[0].latitude,
+                periods.position[0].longitude,
             )
-            latitude = stat["latitude"]
-            longitude = stat["longitude"]
 
-            holes_to_fill = data[
-                data.index.to_series().diff() > data.index.to_series().diff().median()
+            holes_to_fill = data.df[
+                data.df.index.to_series().diff()
+                > data.df.index.to_series().diff().median()
```

Comment:
Could you name this as a variable, so others can easer understand what it compares?